pkg/lib/indexer/indexer.go

```diff
@@ -115,151 +141,201 @@ type DeleteFromIndexRequest struct {
 // DeleteFromIndex is an aggregate API used to generate a registry index image
 // without specific operators
 func (i ImageIndexer) DeleteFromIndex(request DeleteFromIndexRequest) error {
-	// set a temp directory
-	tmpDir, err := ioutil.TempDir("./", tmpDirPrefix)
+	buildDir, outDockerfile, cleanup, err := buildContext(request.Generate, request.OutDockerfile)
+	defer cleanup()
 	if err != nil {
 		return err
 	}
-	defer os.RemoveAll(tmpDir)
 
-	databaseFile, err := i.getDatabaseFile(tmpDir, request.FromIndex)
-	if err != nil {
+	// set a temp directory for unpacking an image
+	// this is in its own function context so that the deferred cleanup runs before we do a docker build
+	// which prevents the full contents of the previous image from being in the build context
+	var databasePath string
+	if err := func () error {
```

Comment:
nit: if this is identical to the previous closure, could we just make it a helper function?